### ISSUE TITLE
Empty vectors are removed when using {:format :off}

### ISSUE THREAD

**Issue by lambdank** [kkinnear/zprint#263]
Hello,

When I run `zprint` on a file with this content
```
;;!zprint {:format :off}

(defn testing [] (+ 1 2 3))
```
The output is
```
;;!zprint {:format :off}

(defn testing  (+ 1 2 3))
```

It looks like empty vectors are removed [here](https://github.com/kkinnear/zprint/blob/f9031d6762e3c7dc557baadc1e9d3a842944cb8c/src/zprint/zprint.cljc#L7679), which seems to cause the error.

**Comment by kkinnear** [comment]
Wow, I'm impressed!  That is indeed exactly where the problem occurred.  Turns out that `concat-no-nil` also fails for empty sequences, returning nil.  Thank you for noticing problem, and particularly for the unprecedented debugging.  It made it much easier to figure out the detailed problem and the solution.  Few people have ever gotten into the details of the internals of zprint other than myself.  And since I've been working on it for a long time, it is often kind of new to me too if I haven't worked recently on a particular area.

This will be in `1.2.5`, and I'll move up releasing it given that this is a very serious bug.  It will still be a couple of weeks, unfortunately, as I'm pretty tied up just now.

**Comment by lambdank** [comment]
That's totally fair. Thank you very much for your help. 

Is there some documentation on how to compile the binary for linux?

**Comment by kkinnear** [comment]
Well, compiling GraalVM binaries for Linux is a bit involved.  I'm hearing the you would like a fix for this sooner than later.

A couple of thoughts:

1. I introduced this bug in `1.2.4`, so if you pick up the Linux binary for `1.2.3`, you should be good to go.  Unless, of course, you need a bug fix introduced in `1.2.4` or some new feature of `1.2.4`.
2. I haven't even pushed the fix to GitHub yet.  I could supply you with an intermediate uberjar pretty easily, if you need something in `1.2.4` and a fix for this problem.  Then you could run it with `java -jar zprint-filter-1.2.5-alpha1 ...` where the `...` is whatever you would put after the `zprintl-1.2.4`.  Of course, you need Java for that, and it will also start up a good bit slower.  It will run at least as fast (or possibly faster) once it starts up as the distributed GraalVM binary.

Part of the problem with you building the Linux version yourself is that the version of GraalVM I use is not even distributed anymore, I don't think.  And I've had varying success in upgrading from one GraalVM version to another.  Usually it goes ok, sometimes it very much does not, and then I'm spending a lot of time figuring out what is going on.

If neither of these things works for you, please let me know.  We have a very new capability in GitHub actions does some tests by building a Linux version of the binary, and I might be able to trigger that and recover the results from wherever they are stashed, though I've never actually done that since this whole thing is only days old.

Thanks!

**Comment by kkinnear** [comment]
I put the build artifact from the GitHub action through some additional testing, and it still looks good.  If you need it, it should work well for you.  Let me know if you want it.

**Comment by lambdank** [comment]
Thank you very much for your answers. I'll try out the printl-1.2.5-alpha1.

**Comment by kkinnear** [comment]
There is a new pre-release, [1.2.5-alpha1](https://github.com/kkinnear/zprint/releases/tag/1.2.5-alpha1) which has the binary which should solve your problem.  I don't usually do alpha releases, so I'm more confident of the actual binary than I am of whether or not I've managed to get GitHub to do what I wanted, so let me know if it doesn't work.  Let me know either way if the binary does or doesn't work for you, if you would.  I'm always up for more testing.  I'll support this pre-release until I actually release 1.2.5, just FYI.

**Comment by kkinnear** [comment]
This is fixed in `1.2.5`.
